### PR TITLE
bluetooth: kconfig: Hide host options if not available

### PR DESCRIPTION
**Pull request by rugeGerritsen**
When performing a combined host and controller build, there is no point
in presenting the option to enable a given host feature if the
controller does not support it. This reduces the list of presented
features to enable/disable.

Signed-off-by: Rubin Gerritsen <rubin.gerritsen@nordicsemi.no>

### CODE REVIEW

**Comment by rugeGerritsen** [comment]
@cvinayak, @Thalley  , I'm facing some CI issues with this PR due to the the fact that the controller doesn't select `BT_CTLR_CENTRAL_ISO_SUPPORT` and `BT_CTLR_PERIPHERAL_ISO_SUPPORT`. 

As tests and samples are built for `nrf52840dk_nrf52840` and `nrf52_bsim`, they include a controller. However, as the controller does not mark these features as supported, the build either fails during configuration or during building.

Could we for example:
* Not build the ISO samples requiring peripheral/central in CI for `nrf52840dk_nrf52840`, but rather `nrf5340dk_nrf5340_cpuapp` or explicitly disable building of the controller.
* Temporarily set `BT_CTLR_CENTRAL_ISO_SUPPORT` and `BT_CTLR_PERIPHERAL_ISO_SUPPORT` in the bsim tests relying on these controller features

I also see that I can skip changing the dependencies for ISO for now, but these are really the ones I want to hide.
I'm open for other suggestions.

**Comment by rugeGerritsen** [comment]
> Can you elaborate on which samples these are?

From what I can see in the CI logs it is the iso_connected_benchmark

**Comment by Thalley** [comment]
> > Can you elaborate on which samples these are?
> 
> From what I can see in the CI logs it is the iso_connected_benchmark

Ah, I see. I think the correct solution is to simply remove `nrf52840dk_nrf52840` from the `platform_allow` list, as the `nrf52840dk_nrf52840` does not support this feature yet.

**Comment by rugeGerritsen** [comment]
> > > Can you elaborate on which samples these are?
> > 
> > 
> > From what I can see in the CI logs it is the iso_connected_benchmark
> 
> Ah, I see. I think the correct solution is to simply remove `nrf52840dk_nrf52840` from the `platform_allow` list, as the `nrf52840dk_nrf52840` does not support this feature yet.

@Thalley , what about the tests?  Are you ok with the following?
> Temporarily set BT_CTLR_CENTRAL_ISO_SUPPORT and BT_CTLR_PERIPHERAL_ISO_SUPPORT in the bsim tests relying on these controller features

**Comment by Thalley** [comment]
> @Thalley , what about the tests? Are you ok with the following?
> 
> > Temporarily set BT_CTLR_CENTRAL_ISO_SUPPORT and BT_CTLR_PERIPHERAL_ISO_SUPPORT in the bsim tests relying on these controller features

I think that is fine.

File: subsys/bluetooth/Kconfig.iso [hunk 1]
```diff
@@ -11,13 +11,15 @@ config BT_ISO
 config BT_ISO_UNICAST
 	bool
 	depends on BT_CONN
+	depends on !BT_CTLR || BT_CTLR_CENTRAL_ISO_SUPPORT || BT_CTLR_PERIPHERAL_ISO_SUPPORT
```
Comment (Thalley) [inline subsys/bluetooth/Kconfig.iso:14]: This line shouldn't be necessary as `BT_ISO_UNICAST` is a hidden config enabled by `BT_ISO_PERIPHERAL` or `BT_ISO_CENTRAL` where you've added the same dependencies

**Review by Thalley — DISMISSED** [review]
LGTM with one minor comment

**Review by Thalley — DISMISSED** [review]

**Comment by Thalley** [comment]
> > > @Thalley , what about the tests? Are you ok with the following?
> > > > Temporarily set BT_CTLR_CENTRAL_ISO_SUPPORT and BT_CTLR_PERIPHERAL_ISO_SUPPORT in the bsim tests relying on these controller features
> > 
> > 
> > I think that is fine.
> 
> Unfortunately that does not work:
> 
> > error: BT_CTLR_CENTRAL_ISO_SUPPORT (defined at subsys/bluetooth/controller/Kconfig:58) is assigned
> > in a configuration file, but is not directly user-configurable (has no prompt).

And `BT_CTLR_CENTRAL_ISO_SUPPORT` isn't enabled for the bsim board? 

Can we just enable `BT_CTLR_CENTRAL_ISO_SUPPORT` for the nrf52_bsim board, and then enable `BT_CTLR_CENTRAL_ISO` in the BSIM tests?

**Review by jhedberg — APPROVED** [review]

**Review by Thalley — APPROVED** [review]

**Pull request merged**